maptivist/src/main/java/com/google/sps/servlets/VoteServlet.java

```diff
@@ -40,50 +40,42 @@
 import java.util.List;
 import java.util.Base64;
 
-
 /** Servlet that handles all my received marker data */
 @WebServlet("/votes")
 
 public final class VoteServlet extends HttpServlet {
     
-    public static int voteCount;
-
+   
     @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
- 
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
On one hand, I applaud the code sharing.

On the other hand, I think its better to have separate GET and POST methods. For a typical server's API design, a GET will never change the server state. A POST usually changes server state (though not always).

So it makes sense to implement GET "/votes" that just returns the current vote count and implement a POST "/votes" that updates the votes. That way, in your HTML/Javascript, if you only want to get the current vote count, it's obvious by only sending a GET. If you want to update the votes, then it's obvious by sending a POST.

The way you have it works, and it has lots of nice code re-use :) so if you just want to merge it in like this, that's fine. But here's an optional suggestion:


1) Rename doPost to something like: `getAndMaybeUpdateVotes(bool update, HttpServletRequest request, HttpServeletResponse)`

2) Add:
```
public void doGet(HttpServeletRequest request, HttpServeletResponse) {
  getAndMaybeUpdateVotes(/* update= */ false, request, response);
}

public void doPost(HttpServeletRequest request, HttpServeletResponse) {
  getAndMaybeUpdateVotes( /* update= */ true, request, response);
}
```